components/script_layout_interface/wrapper_traits.rs

```diff
@@ -308,6 +323,10 @@ pub trait ThreadSafeLayoutNode: Clone + Copy + NodeInfo + PartialEq + Sized {
     /// element style is precomputed, not from general layout itself.
     #[inline]
     fn resolved_style(&self) -> Arc<ServoComputedValues> {
+        if self.is_text_node() {
```

Comment:
I think you should be able to assert here that the node is an element? This is only used for queries and pseudo-elements.